source/auth/auth.rst

```diff
@@ -1290,8 +1285,33 @@ User Provided Callbacks
 ```````````````````````
 
 Drivers MUST allow the user to provide callbacks for token request and
-token refresh.  The token request callback MUST accept the OIDCMechanismServerStep1 structure and return an OIDCRequestTokenResult of
-the form:
+token refresh.  The token request callback must have the following signature:
+
+.. code:: typescript
+
+    function onOIDCRequestToken(clientInfo: OIDCClientInfo, serverInfo: OIDCMechanismServerStep1): OIDCRequestTokenResult
+
+
+Where ``OIDCClientInfo`` is defined as:
+
+.. code:: idl
+
+  OIDCClientInfo:
+    description: "The description of the OIDC client"
+    strict: false
+    fields:
+        principalName:
+            description: "The client identifier provided in the access token,
+            typically the Subject Identifier."
+            type: string
+        timeout_seconds:
+            description: "The number of seconds before a request should time out"
+            optional: true
+        timeout_context:
+            description: "A timeout context variable used to cancel the callback"
+            optional: true
```

Comment:
I know it’s late, but here’s an idea: Let’s version this struct? That way we can prevent using a version of the callback implementation that doesn’t understand potential future requirements that the driver puts on it. (For example, when we drop the `ALLOWED_HOSTS` restriction, the callbacks may need to perform some type of validation of their own.)

```suggestion
            optional: true
        version:
            description: "A version identifying breaking changes in the callback protocol, currently 0"
            type: int
```